src/selection_criteria.rs

```diff
@@ -84,6 +85,7 @@ pub type Predicate = Arc<dyn Send + Sync + Fn(&ServerInfo) -> bool>;
 ///
 /// See the [MongoDB docs](https://docs.mongodb.com/manual/core/read-preference) for more details.
 #[derive(Clone, Debug, PartialEq)]
+#[non_exhaustive]
 pub enum ReadPreference {
```

Comment:
can the cases themselves be marked as non-exhaustive? That way we could add a new filed to e.g. `Nearest` without a breaking change.

I don't think we can do it for `Primary`, although I think that one would only ever take a field if multi-master became a thing, and even then it might not be on `Primary`.

Reply:
The compiler doesn't complain, so I guess they can be. We _could_ do it for primary by defining it as `Primary {}`, but I don't think that's a good idea.

Reply:
Left as-is per discussion about this making them non-constructible

Reply:
I think we may need to take the struct with an enum field approach here to allow for things like hedged reads to be non-breaking. This has the unfortuante side effect of allowing users to specify invalid `ReadPreference`s like is possible with `WriteConcern`, though I don't really know of a way around this.

Reply:
I've updated the ReadPreference API as discussed offline